pages/spec/naming.md

```diff
@@ -0,0 +1,171 @@
+# Module name requirements
+
+:::{note}
+Module naming requirements only apply to packages which are uploaded to a fpm registry; by default, no naming rules are enforced for local fpm projects.
```

Comment:
Why not enforcing it for local packages too? We want this to become a habit.

Reply:
Agree with Ondrej here.

Reply:
The reason is that enforcing them now they'd break everything, starting from building fpm (toml-f does not comply). Maybe we should just warn folks for a while, to give them time to update their packages?